clinvar/src/main/scala/org/broadinstitute/monster/etl/clinvar/ClinvarSplitters.scala

```diff
@@ -0,0 +1,278 @@
+package org.broadinstitute.monster.etl.clinvar
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.values.{SCollection, SideOutput}
+import upack._
+
+import scala.collection.mutable
+
+/** Container for functions used to split apart ClinVar's nested data. */
+object ClinvarSplitters {
+
+  import ClinvarContants._
+
+  /** TODO COMMENT */
+  def getVariation(message: Msg): Option[Msg] =
+    VariationTypes.foldLeft(Option.empty[Msg]) { (acc, subtype) =>
+      acc.orElse(message.obj.remove(subtype).map {
+        case arr @ Arr(many) =>
+          many.foreach(_.obj.update(SubclassKey, subtype))
+          arr
+        case other =>
+          other.obj.update(SubclassKey, subtype)
+          other
+      })
+    }
+
+  /** TODO COMMENT */
+  def collectVariantIds(
+    variantMessage: Msg,
+    output: Option[Msg => Unit]
+  )(getId: Msg => Msg): (Msg, List[Msg], List[Msg]) = {
+    val immediateId = getId(variantMessage)
+    val zero = (List.empty[Msg], List.empty[Msg])
+    val (childIds, descendantIds) =
+      getVariation(variantMessage).fold(zero) {
+        case Arr(children) =>
+          children.foldLeft(zero) {
+            case ((childAcc, descandantsAcc), child) =>
+              val (childId, grandChildIds, deepDescendants) =
+                collectVariantIds(child, output)(getId)
+              (childId :: childAcc, grandChildIds ::: deepDescendants ::: descandantsAcc)
+          }
+        case child =>
+          val (childId, grandChildIds, deepDescendants) =
+            collectVariantIds(child, output)(getId)
+          (List(childId), grandChildIds ::: deepDescendants)
+      }
+    variantMessage.obj.update(ChildrenRef, Arr(childIds: _*))
+    variantMessage.obj.update(DescendantsRef, Arr(childIds ::: descendantIds: _*))
+    output.foreach(_.apply(variantMessage))
+    (immediateId, childIds, descendantIds)
+  }
+
+  /** TODO COMMENT */
+  def splitArchives(fullVcvStream: SCollection[Msg])(implicit coder: Coder[Msg]): (
+    SCollection[Msg],
+    SCollection[Msg],
+    SCollection[Msg],
+    SCollection[Msg],
+    SCollection[Msg]
+  ) = {
+    val rcvOut = SideOutput[Msg]
+    val variationOut = SideOutput[Msg]
```

Comment:
Note to self: Since this step always outputs a variation, but only sometimes outputs a VCV, it'd make more sense to have the VCV be a side output and the "main" output be for variations.